src/useSubscription.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  ApolloClient,
+  ApolloError,
+  OperationVariables,
+  SubscriptionOptions,
+} from 'apollo-client';
+import { DocumentNode } from 'graphql';
+import { useEffect, useRef, useState } from 'react';
+import isEqual from 'react-fast-compare';
```

Comment:
`react-fast-compare` isn't currently a dependency, can we get this from an existing dependency, or should do you want to add `react-fast-compare` as a dependency?